tests/core/pyspec/eth2spec/test/lightclient_patch/block_processing/test_process_sync_committee.py

```diff
@@ -163,24 +232,26 @@ def test_invalid_signature_previous_committee(spec, state):
     # To get a distinct committee so we can generate an "old" signature, we need to advance
     # 2 EPOCHS_PER_SYNC_COMMITTEE_PERIOD periods.
     current_epoch = spec.get_current_epoch(state)
-    previous_committee = state.next_sync_committee
 
     epoch_in_future_sync_commitee_period = current_epoch + 2 * spec.EPOCHS_PER_SYNC_COMMITTEE_PERIOD
     slot_in_future_sync_committee_period = epoch_in_future_sync_commitee_period * spec.SLOTS_PER_EPOCH
     transition_to(spec, state, slot_in_future_sync_committee_period)
 
+    # Create incorrect_committee for generating invalid signature.
     pubkeys = [validator.pubkey for validator in state.validators]
-    committee = [pubkeys.index(pubkey) for pubkey in previous_committee.pubkeys]
+    correct_committee = [pubkeys.index(pubkey) for pubkey in state.current_sync_committee.pubkeys]
+    incorrect_committee = [(correct_committee[0] + 1) % len(pubkeys)] + correct_committee[1:]
+    assert correct_committee != incorrect_committee
```

Comment:
@ralexstokes 

I rewrote it a bit so that the signing committee must be incorrect.

Reply:
if we do this change, then the test case becomes very similar to the condition tested in `test_invalid_signature_extra_participant`.

the thinking behind this test was to explicitly test that using the `previous_committee` from the `BeaconState` was invalid in the event that an eth2 client had a bug in caching committees, or something like this.

what do you think about reverting it to how it was and just adding an `assert committee != state.next_sync_committee.pubkeys`?

this would still capture the aim of the original test although would still fail intermittently when we end up with a duplicate committee.  which suggests we need a facility across this module for sampling committees w/ or w/o duplication like @djrtwo was suggesting....

let me add this to see if it adds a bunch of overhead to the test execution

Reply:
Oh sorry I didn't know it meant to test the caching committees. But isn't the previous committee `old_sync_committee`

I changed it back now plus assertion and `@with_configs([MINIMAL], reason="to produce different committee sets")`. With mainnet config, the sampled sync committee is usually the same group since we only have 256 active validators.

Yeah it's still possible that the `set(old_sync_committee.pubkeys) == set(state.current_sync_committee.pubkeys)` and it seems more inefficient to find the suitable committee than #2189... IMO it's okay to keep the status quo and only run with the minimal config.